zvariant/src/signature.rs

```diff
@@ -294,6 +294,29 @@ impl<'a> Signature<'a> {
         clone
     }
 
+    /// Associated function to check whether the signature string slice has
+    /// balanced parentheses.   
+    pub fn has_balanced_parentheses(signature_str: &str) -> bool {
```

Comment:
* why is this public? It's not even a method of Signature. Its just a private utility function so should be private and separate function.
* I thought we don't need it?

Reply:
?

Reply:
You addressed the first point but didn't answer my question. IIRC (I've to admit it's hard to remember our last convo on this) we had said we don't need to check for balancing?

Reply:
Looking at the code now and previous comment, I get it. However, I don't think it deserves to be an associated function. Please just move it to be a separate function. Then I believe we're good to merge. :)